### PR TITLE
[Audio] Config option to not disconnect when just bot in voice

### PR DESCRIPTION
**Pull request by FrostTheFox**
Little too hard to explain in the title, but adds an option in audio settings to enable or disable the check that's basically "if only bot in here, disconnect in 5 minutes"

Python noob, be nice :3

also i didn't know what would be a good variable, if you have a better idea lemme know

### CODE REVIEW

File: cogs/audio.py [hunk 1]
```diff
@@ -290,6 +290,9 @@ def _cache_max(self):
         setting_max = self.settings["MAX_CACHE"]
         return max([setting_max, self._cache_min()])  # enforcing hard limit
 
+    def _noppl_disconnect(self):
+        setting_nodis = self.settings["NOPPL_DIS"]
```
Comment (Chovin) [inline cogs/audio.py]: `_noppl_disconnect ` is never called and even if it was, the setting is never put into the settings dict `self.settings` or into the `settings.json` file.

File: cogs/audio.py [hunk 2]
```diff
@@ -1849,10 +1852,15 @@ def currently_downloading(self, server):
                     stop_times[server] = int(time.time())
 
                 if hasattr(vc, 'audio_player'):
-                    if (vc.audio_player.is_done() or len(vc.channel.voice_members) == 1):
+                    if (vc.audio_player.is_done() == 1):
```
Comment (Chovin) [inline cogs/audio.py]: `== 1` eh?

File: cogs/audio.py [hunk 3]
```diff
@@ -1849,10 +1852,15 @@ def currently_downloading(self, server):
...
                         if server not in stop_times or stop_times[server] is None:
                             log.debug("putting sid {} in stop loop".format(server.id))
                             stop_times[server] = int(time.time())
+                    elif self._noppl_disconnect == False:
```
Comment (Chovin) [inline cogs/audio.py]: you defined this as a function above. this isn't how you call a function.
`if not a_boolean:` is preferred over `if a_boolean == False:`

**Review by Chovin — CHANGES_REQUESTED** [review]
thanks for your work.
I suggest you look at some simple cogs for examples, there are some pretty useful examples in some of them.

File: cogs/audio.py (hunk 3 above)
Comment (FrostTheFox) [reply cogs/audio.py]: whoop i'm an idiot

File: cogs/audio.py (hunk 2 above)
Comment (FrostTheFox) [reply cogs/audio.py]: haha understood 👍

File: cogs/audio.py (hunk 1 above)
Comment (FrostTheFox) [reply cogs/audio.py]: Wouldn't the addition at line 2060 add it to settings.json if not present? Just tried and it seemed to...

**Comment by FrostTheFox** [comment]
Alright, it's at least working now.

File: cogs/audio.py (hunk 2 above)
Comment (FrostTheFox) [reply cogs/audio.py]: Fixed this 👍

File: cogs/audio.py (hunk 3 above)
Comment (FrostTheFox) [reply cogs/audio.py]: fixed 👍

File: cogs/audio.py (hunk 1 above)
Comment (tekulvw) [reply cogs/audio.py]: Yeah it will, Kowlin pulled some magic out of his ass a while back on load to make this happen

**Comment by tekulvw** [comment]
Can you make this editable by contributors? You've got a good start, there's just a few things I'd like to change and add real quick

**Comment by FrostTheFox** [comment]
Done. Sorry, it became habit on other projects.

**Comment by tekulvw** [comment]
Okay I think we're ready to go now @Twentysix26

**Comment by Twentysix26** [comment]
As it is, it's a global setting. It has no reason to be global.

**Review by Twentysix26 — CHANGES_REQUESTED** [review]
It should be server based

**Review by Twentysix26 — APPROVED** [review]

**Comment by FrostTheFox** [comment]
heh, thanks for fixing this up, I wrote this really late at night and I may have been drinking some wine so I was not at peak performance :P

**Pull request merged**